### PR TITLE
Developer docs for PartitionConsumer and PartitionAcceptor

### PR DESCRIPTION
**Pull request by anwar6953**
Describing how to use PartitionConsume in conjunction with PartitionAcceptor.
Including a code sample in the develop docs, after describing it.

Marking the relevant classes as Beta (PartitionedFileSet itself is Beta).

https://issues.cask.co/browse/CDAP-5534

Building [Quick Build 3](http://builds.cask.co/browse/CDAP-DOB186-3)

### CODE REVIEW

File: cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst [hunk 1]
```diff
@@ -356,7 +356,7 @@ will be registered with the output ``PartitionedFileSet`` at the end of the job.
 Note that any partitions written to must not previously exist. Otherwise, the MapReduce job will fail at the
 end of the job and none of the partitions will be added to the ``PartitionedFileSet``.
 
-Incrementally Processing PartitionedFileSets
+Incrementally Processing PartitionedFileSets (MapReduce)
```
Comment (anew) [inline cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst]: Not sure about the parentheses here and below. @john can you find a better way?

Comment (anwar6953) [inline cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst]: @awholegunch is going to break it up into a broader section for Incremental processing and then have these two as sub-sections of that.

File: cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst [hunk 2]
```diff
@@ -390,6 +390,62 @@ This ensures that the next time the MapReduce job runs, it processes only the ne
...
+  final List<PartitionDetail> partitions = partitionConsumer.consumePartitions().getPartitions();
+
+  // process partitions
+
+  // once done processing, onFinish must be called with a boolean value indicating success or failure, so that
+  // the partitions' can be marked accordingly for completion or retries in the future
+  partitionConsumer.onFinish(partitions, true);
+
+The ``consumePartitions`` method of the PartitionConsumer can optionally take in a limit (int), which will
+limit the number of returned partitions. It also also take in a PartitionAcceptor, which allows the user to
```
Comment (anew) [inline cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst]: Also also

Comment (anew) [inline cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst]: Which allows you

File: cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst [hunk 3]
```diff
@@ -390,6 +392,61 @@ This ensures that the next time the MapReduce job runs, it processes only the ne
...
+  // This can be in any program where we have access to Datasets,
+  // such as a Worker, Workflow Action, or even in a MapReduce
+  PartitionConsumer consumer =
+    new ConcurrentPartitionConsumer(partitionedFileSet, new CustomStatePersistor(persistenceTable));
+
+  // Call consumePartitions to get a list of partitions to process
+  final List<PartitionDetail> partitions = partitionConsumer.consumePartitions().getPartitions();
+
+  // Process partitions
+
```
Comment (awholegunch) [inline cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst]: does something go in here? If so, indicate with ellipsis:
```
. . .

Comment (anwar6953) [inline cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst]: Yea, it was meant to indicate any custom processing by the user of these partitions.

File: cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst [hunk 4]
```diff
@@ -390,6 +392,61 @@ This ensures that the next time the MapReduce job runs, it processes only the ne
     partitionCommitter.onFinish(succeeded);
   }
 
+Processing using a Generic
```
Comment (awholegunch) [inline cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst]: I'm not crazy about this title, as *Generic* really refers to something else...any suggestions?

What about "Processing using a PartitionConsumer"?

Comment (anwar6953) [inline cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst]: yea, that sounds good.

**Comment by awholegunch** [comment]
LGTM. I made one minor change. Running a final build, #4.

**Comment by anew** [comment]
LGTM

**Comment by anwar6953** [comment]
Squashed commits and merging.
Not running unit tests, as it is just documentation updates.

**Pull request merged**